pkg/deployer/container/container_reconcile.go

```diff
@@ -178,44 +178,50 @@ func (c *Container) Reconcile(ctx context.Context, operation container.Operation
 		return nil
 	}
 
-	if pod == nil {
-		return nil
-	}
 	operationName := "Complete"
+	if pod != nil {
+		podSucceeded := pod.Status.Phase == corev1.PodSucceeded
+		if podSucceeded {
+			if err := c.SyncExport(ctx); err != nil {
+				return lserrors.NewWrappedError(err,
+					operationName, "SyncExport", err.Error())
+			}
+		} else if pod.Status.Phase == corev1.PodFailed {
+			lsv1alpha1helper.SetDeployItemToFailed(c.DeployItem)
+		}
 
-	if pod.Status.Phase == corev1.PodSucceeded {
-		if err := c.SyncExport(ctx); err != nil {
+		c.ProviderStatus.LastOperation = string(operation)
+		if err := c.collectAndSetPodStatus(pod, podSucceeded); err != nil {
 			return lserrors.NewWrappedError(err,
-				operationName, "SyncExport", err.Error())
+				"Reconcile", "UpdatePodStatus", err.Error())
 		}
-		c.DeployItem.Status.Phase = lsv1alpha1.DeployItemPhases.Succeeded
-	}
-	if pod.Status.Phase == corev1.PodFailed {
-		lsv1alpha1helper.SetDeployItemToFailed(c.DeployItem)
-	}
 
-	c.ProviderStatus.LastOperation = string(operation)
-	if err := c.collectAndSetPodStatus(pod); err != nil {
-		return lserrors.NewWrappedError(err,
-			"Reconcile", "UpdatePodStatus", err.Error())
-	}
+		// write status to ensure podStatus is saved before deleting the pod
+		if err := lsWriter.UpdateDeployItemStatus(ctx, read_write_layer.W000031, c.DeployItem); err != nil {
+			return lserrors.NewWrappedError(err, operationName, "UpdateDeployItemStatus", err.Error())
+		}
 
-	if err := lsWriter.UpdateDeployItemStatus(ctx, read_write_layer.W000066, c.DeployItem); err != nil {
-		return lserrors.NewWrappedError(err,
-			operationName, "UpdateDeployItemStatus", err.Error())
+		// only remove the finalizer if we get the status of the pod
+		logger.Debug("Deleting pod, as it has finished", "podStatus", pod.Status.Phase)
+		if err := c.CleanupPod(ctx, pod); err != nil {
+			return err
+		}
 	}
-
-	// only remove the finalizer if we get the status of the pod
-	if err := c.CleanupPod(ctx, pod); err != nil {
-		return err
+	if c.ProviderStatus != nil && c.ProviderStatus.PodStatus != nil && c.ProviderStatus.PodStatus.LastSuccessfulJobID != nil && *c.ProviderStatus.PodStatus.LastSuccessfulJobID == c.DeployItem.Status.JobID {
```

Comment:
I guess `*c.ProviderStatus.PodStatus.LastSuccessfulJobID == c.DeployItem.Status.JobID` must not be required here. This could not happen otherwise we made a mistake in our model. The reconcile method should only be called if the jod ids differ.

Reply:
Sorry my mistake - I did not get it right :-)